include/scale/scale.hpp

```diff
@@ -35,6 +33,20 @@
   SCALE_EMPTY_DECODER(TargetType)
 
 namespace scale {
+  template <typename F>
+  outcome::result<std::invoke_result_t<F>> outcomeCatch(F &&f) {
+    try {
+      if constexpr (std::is_void_v<std::invoke_result_t<F>>) {
+        f();
+        return outcome::success();
+      } else {
+        return outcome::success(f());
+      }
+    } catch (std::system_error &e) {
```

Comment:
If it's meant to catch exceptions thrown by SCALE, maybe make a subclass for SCALE exceptions for clarity?

Reply:
No, it catches any `outcome::result` errors thrown by user decoders